src/_guides/language/effective-dart/design.md

```diff
@@ -1045,48 +1040,43 @@ confusion, we avoid saying "untyped" and instead use the following terminology:
 
-Sometimes, Dart infers a type, but not the type you want. For example, you may
-want a variable's type to be a supertype of the initializer's type so that you
-can later assign some other sibling type to the variable:
+The guidelines here strike the best balance we've found between brevity and
+explicitness, flexibility and safety. When deciding which types to write, you
+need to answer two questions:
 
-{:.good-style}
-<?code-excerpt "misc/lib/effective_dart/design_good.dart (inferred-wrong)"?>
-{% prettify dart %}
-num highScore(List<num> scores) {
-  num highest = 0;
-  for (var score in scores) {
-    if (score > highest) highest = score;
-  }
-  return highest;
-}
-{% endprettify %}
+* What types should I write because I think it's best for them to be visible in
+  the code?
 
-{:.bad-style}
-<?code-excerpt "misc/lib/effective_dart/design_bad.dart (inferred-wrong)" replace="/ +\/\/ ignore: invalid_assignment\n//g"?>
-{% prettify dart %}
-num highScore(List<num> scores) {
-  var highest = 0;
-  for (var score in scores) {
-    if (score > highest) highest = score;
-  }
-  return highest;
-}
-{% endprettify %}
+* Which types should I write because inference can't provide them for me?
 
-Here, if `scores` contains doubles, like `[1.2]`, then the assignment to
-`highest` will fail since its inferred type is `int`, not `num`. In these cases,
-explicit annotations make sense. In the guidelines below, treat *inferred* as
-meaning *inferred to be the type you want*.
+These guidelines help you answer the first question:
+
+* [PREFER type annotating public top-level variables and fields if the type isn't obvious.](#prefer-type-annotating-public-top-level-variables-and-fields-if-the-type-isnt-obvious)
+* [CONSIDER type annotating private top-level variables and fields if the type isn't obvious.](#consider-type-annotating-private-top-level-variables-and-fields-if-the-type-isnt-obvious)
+* [AVOID type annotating initialized local variables.](#avoid-type-annotating-initialized-local-variables)
+* [AVOID annotating inferred parameter types on function expressions.](#avoid-annotating-inferred-parameter-types-on-function-expressions)
+* [AVOID redundant type arguments on generic invocations.](#avoid-redundant-type-arguments-on-generic-invocations)
+
+These cover the second:
+
+* [DO annotate when Dart infers the wrong type.](#do-annotate-when-dart-infers-the-wrong-type)
+* [PREFER annotating with `dynamic` instead of letting inference fail.](#prefer-annotating-with-dynamic-instead-of-letting-inference-fail)
+
+The remaining guidelines cover other more specific questions around types.
 
 
-### DO type annotate public declarations whose type isn't inferred.
+### PREFER type annotating public top-level variables and fields if the type isn't obvious.
 
-Type annotations are important documentation for how a library should be used.
-They form boundaries between regions of a program to isolate the source of a
-type error.
+Consider:
```

Comment:
This should be a sentence explaining what to consider.

Also... this doesn't look like a variable or field to me. What does this section really cover?

Reply:
Oops. Moved some stuff around and left this hanging. Fixed.